### PR TITLE
Added details for Testbed Repo

### PR DESCRIPTION
**Pull request by EdZou**
**Description:** <Describe what has changed. 
Modified README.md for testbed repo

**Documentation:**
1. Added pipeline diagram for testbed and description for loadGenerator and Mockbackend
2. Added step-by-step instruction for development and test running.

### CODE REVIEW

File: testbed/README.md [hunk 1]
```diff
@@ -13,6 +13,12 @@ a global instance of `testbed.TestResultsSummary` to it.
 Each test case within the suite should create a `testbed.TestCase` and supply implementations of each of the various
 interfaces the `NewTestCase` function takes as parameters.
 
+Within `testbed.TestCase`, it uses `LoadGenerator` and `MockBackend` to further encapsulate pluggable components. `LoadGenerator` further encapsulates `DataProvider` and `DataSender` in order to generate and send  data.  `MockBackend` further encapsulate `DataReceiver` and provide comsume functionality.
```
Comment (anuraaga) [inline testbed/README.md:16]:
```suggestion
`testbed.TestCase` uses `LoadGenerator` and `MockBackend` to further encapsulate pluggable components. `LoadGenerator` further encapsulates `DataProvider` and `DataSender` in order to generate and send  data.  `MockBackend` further encapsulate `DataReceiver` and provide comsume functionality.
```

File: testbed/README.md [hunk 2]
```diff
@@ -13,6 +13,12 @@ a global instance of `testbed.TestResultsSummary` to it.
...
+
+For instance, if using the existing end-to-end test, the general pipeline can be (Note that MockBackend does not really have a consumer instance, only to make it intuitive, this diagram draws a separate module):
```
Comment (anuraaga) [inline testbed/README.md:18]:
```suggestion
For instance, if using the existing end-to-end test, the general pipeline can be (Note that MockBackend does not really have a consumer instance, only to make it intuitive, this diagram draws it a separate module):
```

File: testbed/README.md [hunk 3]
```diff
@@ -39,3 +45,93 @@ interfaces the `NewTestCase` function takes as parameters.
 * `TestResultsSummary` - Records itemized test case results plus a summary of one category of testing.
   * `PerformanceResults` - Implementation of `TestResultsSummary` with fields suitable for reporting performance test results.
   * `CorrectnessResults` - Implementation of `TestResultsSummary` with fields suitable for reporting data translation correctness test results.
+
+## Step-by-Step Instruction for development
```
Comment (anuraaga) [inline testbed/README.md:49]:
```suggestion
## Adding new receivers and/or exporters to the testbed
```

File: testbed/README.md [hunk 4]
```diff
@@ -39,3 +45,93 @@ interfaces the `NewTestCase` function takes as parameters.
...
+
+Generally, when designing a test for new exporter and receiver components, by taking use of existing implementation of testbed, developers should mainly focus on designing and implementing the components with yellow background in the diagram above:
```
Comment (anuraaga) [inline testbed/README.md:51]:
```suggestion
Generally, when designing a test for new exporter and receiver components, developers should mainly focus on designing and implementing the components with yellow background in the diagram above as the other components are implemented by the testbed framework :
```

Comment (EdZou) [reply testbed/README.md:51]: > Any reason for the Untitled.md?

Sorry, my fault, I will tackle it! Thank you!

Comment (EdZou) [reply testbed/README.md]: Thank you for your suggestions @anuraaga , I have already revised the content and made new commits!

**Comment by EdZou** [comment]
> Any reason for the Untitled.md?
Sorry for the miss, already removed it, thank you so much! @bogdandrutu

**Comment by mtwo** [comment]
@bogdandrutu can you re-review, as you already commented? We discussed this in the Collector SIG meeting today

File: testbed/README.md [hunk 5]
```diff
@@ -13,6 +13,12 @@ a global instance of `testbed.TestResultsSummary` to it.
 Each test case within the suite should create a `testbed.TestCase` and supply implementations of each of the various
 interfaces the `NewTestCase` function takes as parameters.
 
+`testbed.TestCase` uses `LoadGenerator` and `MockBackend` to further encapsulate pluggable components. `LoadGenerator` further encapsulates `DataProvider` and `DataSender` in order to generate and send  data.  `MockBackend` further encapsulate `DataReceiver` and provide comsume functionality.
```
Comment (pmcollins) [inline testbed/README.md:16]: I wonder if it makes sense to put this in a `## Data Flow` section and talk about how data is generated by a `LoadGenerator` and consumed by a `MockBackend`, then point out how those two objects/structs wrap a dataProvider+sender and a receiver, respectively.

Comment (EdZou) [reply testbed/README.md:16]: Wow, sounds great! Let me modify it and make another commit, thank you!

**Pull request merged**